staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go

```diff
@@ -256,6 +389,44 @@ func (w *WebhookAuthorizer) RulesFor(user user.Info, namespace string) ([]author
 	return resourceRules, nonResourceRules, incomplete, fmt.Errorf("webhook authorizer does not support user rule resolution")
 }
 
+// Match is used to evaluate the SubjectAccessReviewSpec against
+// the authorizer's matchConditions in the form of cel expressions
+// to return match or no match found, which then is used to
+// determine if the webhook should be skipped.
+func (w *WebhookAuthorizer) Match(ctx context.Context, r *authorizationv1.SubjectAccessReview) (bool, error) {
+	// An empty list of matchConditions matches all requests.
+	if len(w.matchConditions) == 0 || w.celMapper == nil {
+		return true, nil
+	}
+	evalResults, err := w.celMapper.eval(ctx, r)
+	if err != nil {
+		return false, fmt.Errorf("authz: cel program eval failed: %v", err)
+	}
+	// If at least one matchCondition evaluates to FALSE, then the webhook is skipped.
+	// If ALL matchConditions evaluate to TRUE, then the webhook is called.
+	for _, evalResult := range evalResults {
+		if evalResult.Error != nil {
+			return false, fmt.Errorf("authz: error evaluating SubjectAccessReview expression: %v", evalResult.Error)
+		}
+		match := evalResult.EvalResult.Value().(bool)
```

Comment:
I think these types of error should have been caught by https://github.com/kubernetes/kubernetes/pull/121223/files#diff-32d8c895ea9e6f4bf62bbf2927bc040fede98fd7e46e57b0c233f566e68611ecR97 at compile time. But I have added the type check at eval time just in case.